src/averageAges.js

```diff
@@ -36,8 +45,22 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = !!withChildren
+    ? people.filter(person =>
+      person.sex === 'f' && people.some(child => child.mother === person.name))
+    : people.filter(person => person.sex === FEMALE);
+
+  const womenTotalAge = women.reduce((acc, woman) => {
+    return acc + (woman.died - woman.born);
+  }, 0);
+
+  womenTotalAge.sort().reverse();
+
+  const averafeWoomanAge = womenTotalAge / women.length + 1;
```

Comment:
There is a typo in the variable name `averafeWoomanAge`; it should be `averageWomanAge`. Also, adding 1 to the average age calculation (`womenTotalAge / women.length + 1`) is incorrect; the average should be calculated without this adjustment.